cucoslib/workers/recommender.py

```diff
@@ -59,15 +59,26 @@ def str_value_cleaner(safe_string):
             result = pattern_n2_remove.sub('', temp_str)
         return result
 
+    def execute_dsl(self, payload):
+        max_retry = int (os.getenv('MAX_GREMLIN_RETRY_COUNT', '3'))
+        for i in range(max_retry):
```

Comment:
I think `requests` library supports retries on its own. I haven't used the functionality myself yet, but using the existing, well-tested, solution rather than our own could simplify the code a bit.